R/estimate_truncation.R

```diff
@@ -193,6 +197,13 @@ estimate_truncation <- function(obs, max_truncation, trunc_max = 10,
 
   # combine into ordered matrix
   dirty_obs <- purrr::map(obs, data.table::as.data.table)
+  dirty_obs <- purrr::map(dirty_obs,
+    create_clean_reported_cases,
+      horizon = 0,
+      filter_leading_zeros = filter_leading_zeros,
+      zero_threshold = zero_threshold,
+      add_breakpoints = FALSE
+  )
```

Comment:
```suggestion
  )
earliest_date <- max(
  as.Date(
    purrr::map_chr(dirty_obs, function(x) x[, as.character(min(date))])
  )
)
dirty_obs <- purrr::map(dirty_obs, function(x) x[date >= earliest_date])
```
This would make sure the data all align on the earliest date even if some zeroes are removed from some of the data frames in the list.

Reply:
Hmmmm strange. I fixed and pushed this earlier. 🤔  I was doing something like

```r
obs <- purrr::map(dirty_obs, function(dt) {
    dt_merged <- merge.data.table(
      dt,
      dirty_obs[[length(dirty_obs)]],
      by = c("date", "confirm"),
      all.y = TRUE
    )
    dt_merged[date <= max(dt$date)]
  })
```

I think it's probably not efficient though. Yours is better.

Reply:
I've made this change here 58ee99604243d71ee989677bbf1968c9adee6dc4.